handlers/slot.go

```diff
@@ -205,14 +206,15 @@ func getAttestationsData(slot uint64, onlyFirst bool) ([]*types.BlockPageAttesta
 }
 
 func GetSlotPageData(blockSlot uint64) (*types.BlockPageData, error) {
+	latestFinalizedEpoch := services.LatestFinalizedEpoch()
 	blockPageData := types.BlockPageData{}
 	blockPageData.Mainnet = utils.Config.Chain.Config.ConfigName == "mainnet"
 	// for the first slot in an epoch the previous epoch defines the finalized state
 	err := db.ReaderDb.Get(&blockPageData, `
 		SELECT
 			blocks.epoch,
-			COALESCE(epochs.finalized, false) AS epoch_finalized,
-			COALESCE(prev_epoch.finalized, false) AS prev_epoch_finalized,
+			(COALESCE(epochs.epoch, 0) <= $3) AS epoch_finalized,
+			(GREATEST((blocks.slot-1)/$2-1,0) <= $3) AS prev_epoch_finalized,
```

Comment:
**issue:** Also the case here I think. Removing the left join removes the possibility that it could return no rows which before would result in a **false** for the **prev_epoch_finalized**.
Now the behaviour could be different.

Please think this through again.
If you say that there are no problems it's fine with me.

Reply:
Like the other comment + it's more performant this way.